Tweaks to allow Calibre to use the older version of python3-py7zr in noble.

--- a/src/calibre/utils/seven_zip.py
+++ b/src/calibre/utils/seven_zip.py
@@ -32,9 +32,8 @@ class Writer:
 
 
 def read_file(archive, name):
-    w = Writer()
-    archive.extract(targets=[name], factory=w)
-    for v in w.outputs.values():
+    d = archive.read(targets=[name])
+    for v in d.values():
         return v.getvalue()
     raise KeyError(f'No file named {name} in archive')
 
@@ -101,9 +100,8 @@ def test_basic():
         with open_archive(os.path.join('a.7z')) as zf:
             if set(zf.getnames()) != set(tdata):
                 raise ValueError('names not equal')
-            w = Writer()
-            zf.extractall(factory=w)
-            read_data = w.asdatadict()
+            d = zf.readall()
+            read_data = { x:y.getvalue() for x,y in d.items() }
             if read_data != tdata:
                 raise ValueError('data not equal')
 
